invenio_deposit/views/rest.py

```diff
@@ -299,15 +308,18 @@ def get(self, pid, record, key, version_id, **kwargs):
     @need_record_permission('update_permission_factory')
     def put(self, pid, record, key):
         """Handle PUT deposit files."""
-        data = json.loads(request.data.decode('utf-8'))
-        new_key = data['filename']
-        if not new_key:
+        try:
+            data = json.loads(request.data.decode('utf-8'))
+            new_key = data['filename']
+        except KeyError:
             abort(400)
-        if key in record.files:
+        if not new_key or re.match(".*\/\.\.\/.*", new_key):
```

Comment:
@hachreak @lnielsen Shall we use `secure_filename` or other global validation for `key`?

--
http://werkzeug.pocoo.org/docs/0.11/utils/#werkzeug.utils.secure_filename